### ISSUE TITLE
"subnets" input order differs from "subnets_self_links" output order

### ISSUE THREAD

**Issue by tuunit** [terraform-google-modules/terraform-google-network#400]
### TL;DR

Unfortunately, adding subnets is handled via a list input but the output is generated from a map inside the subnets module. Which is lexicographically ordered. Therefore using the same index for the subnets input list and the subnets_self_links list don't necessarily reflect the same subnets.

### Expected behavior

The `subnets_self_links` output should always reflect the same order as the `subnets` input order.

### Observed behavior

The `subnets_self_links` output is ordered based on the name of the subnets instead of the order of the `subnets` input list order.

This problem occurs because of the transformation inside the subnet module.
https://github.com/terraform-google-modules/terraform-google-network/blob/master/modules/subnets/main.tf

`
...

locals {
  subnets = {
    for x in var.subnets :
    "${x.subnet_region}/${x.subnet_name}" => x
  }
}

resource "google_compute_subnetwork" "subnetwork" {
  for_each                 = local.subnets

...
`

Unfortunately, maps are iterated based on the keys in a lexicographical order and therefore the resulting `subnets_self_links` order is dependent on the lexicographic order instead of the input list order.

### Terraform Configuration

```hcl
module "common_vpc" {
  source  = "terraform-google-modules/network/google"
  version = "~> 5.2"

  project_id   = module.project_network.project_id
  network_name = "vpc-${local.client}-host"
  routing_mode = "REGIONAL"

  # Will result in index 0 for sub-vpc-a and index 1 for sub-vpc-b in the subnets_self_links output
  subnets = [
    {
      subnet_name           = "sub-vpc-${local.client}-b"
      subnet_ip             = "10.0.0.0/25"
      subnet_region         = local.default_region
      subnet_private_access = true
    },
    {
      subnet_name           = "sub-vpc-${local.client}-a"
      subnet_ip             = var.subnet_ip
      subnet_region         = local.default_region
      subnet_private_access = true
    }
  ]

  routes = []
}
```

### Terraform Version

```sh
v1.3.0
```
